src/databricks/labs/ucx/hive_metastore/locations.py

```diff
@@ -248,3 +250,109 @@ def _try_fetch(self) -> Iterable[Mount]:
+
+        Returns:
+            list[Table]: A list of Table objects representing the snapshot of tables.
+        """
+        return self._snapshot(partial(self._try_load), partial(self._crawl))
+
+    def _try_load(self) -> Iterable[Table]:
+        """Tries to load table information from the database or throws TABLE_OR_VIEW_NOT_FOUND error"""
+        for row in self._fetch(f"SELECT * FROM {escape_sql_identifier(self.full_name)}"):
+            yield Table(*row)
+
+    def _crawl(self):
+        all_mounts = self._mc.snapshot()
+        all_tables = []
+        for mount in all_mounts:
+            if self._include_mounts and mount.name not in self._include_mounts:
+                continue
+            table_paths = self._find_delta_log_folders(mount.name)
+            for path, entry in table_paths.items():
+                guess_table = os.path.basename(path)
+                table = Table(
+                    catalog="hive_metastore",
+                    database=f"mounted_{mount.name}",
+                    name=guess_table,
+                    object_type="EXTERNAL",
+                    table_format=entry.format,
+                    location=path,
+                    storage_properties="PARTITIONED" if entry.is_partitioned else None,
+                )
+                all_tables.append(table)
+        return all_tables
+
+    def _find_delta_log_folders(self, root_dir, delta_log_folders=None) -> dict:
+        if delta_log_folders is None:
+            delta_log_folders = {}
+
+        entries = self._dbutils.fs.ls(root_dir)
+        for entry in entries:
+            if self._is_irrelevant(entry.name):
+                continue
+
+            path = os.path.dirname(entry.path)
+
+            if self._path_is_delta(delta_log_folders, path):
+                continue
+
+            if entry.name == "_delta_log":
+                logger.debug(f"Found delta table {path}")
+                if delta_log_folders.get(path) and delta_log_folders.get(path).is_partitioned:
+                    delta_log_folders[path] = TableInMount(format="DELTA", is_partitioned=True)
+                else:
+                    delta_log_folders[path] = TableInMount(format="DELTA", is_partitioned=False)
+            elif self._is_partitioned(entry.path):
+                delta_log_folders[path] = TableInMount(format="PARQUET", is_partitioned=True)
+            elif self._is_parquet(entry.path):
+                delta_log_folders[path] = TableInMount(format="PARQUET", is_partitioned=False)
+            else:
+                self._find_delta_log_folders(entry.path, delta_log_folders)
+
+        return delta_log_folders
+
+    def _path_is_delta(self, delta_log_folders, path: str) -> bool:
+        return delta_log_folders.get(path) and delta_log_folders.get(path).format == "DELTA"
+
+    def _is_partitioned(self, entry: str) -> bool:
+        partitioned_pattern = {'='}
+        return any(pattern in entry for pattern in partitioned_pattern)
```

Comment:
Isn't just below line would suffice?

```suggestion
    def _is_partitioned(self, entry: str) -> bool:
        return '=' in entry
```